shell/shared/grain.js

```diff
@@ -540,12 +546,58 @@ if (Meteor.isClient) {
     },
   });
 
+  Template.shareWithOthers.onRendered(function () {
+    this.find("[role=tab]").focus();
+  });
+  var activateTargetTab = function(event, instance) {
+    // Deactivate all tabs and all tab panels.
+    instance.findAll("ul[role=tablist]>li[role=tab]").forEach(function (element) {
+      element.setAttribute("aria-selected", false);
+    });
+    instance.findAll(".tabpanel").forEach(function (element) {
+      element.setAttribute("aria-hidden", true);
+    });
+
+    // Activate the tab header the user selected.
+    event.currentTarget.setAttribute("aria-selected", true);
+    // Show the corresponding tab panel.
+    var idToShow = event.currentTarget.getAttribute("aria-controls");
+    var tabPanelToShow = instance.find("#" + idToShow);
+    tabPanelToShow.setAttribute("aria-hidden", false);
+  };
   Template.shareWithOthers.events({
-    "click .shareable-link": function (event, instance) {
-      instance.find(".share-tabs").setAttribute("data-which-tab", "shareable-link");
-    },
-    "click .send-invite": function (event, instance) {
-      instance.find(".share-tabs").setAttribute("data-which-tab", "send-invite");
+    "click #send-invite-tab-header": activateTargetTab,
+    "click #shareable-link-tab-header": activateTargetTab,
+    "keydown [role=tab]": function(event, template) {
+      if (event.keyCode == 38 || event.keyCode == 40) { // up and down arrows
```

Comment:
Not really.  Also, the IDs exist because the aria-labelfor and aria-labelledby attributes requires IDs.

I'd like to (at some point) make this into a package that handles things more declaratively, and has an array of tabs and tab headers, and can operate over them more generically.  But I figured we could build that when we have more than one tab widget.